druid/src/env.rs

```diff
@@ -342,13 +342,20 @@ impl Env {
     ) -> Result<(), ValueTypeError> {
         let env = Arc::make_mut(&mut self.0);
         let key = key.into();
-        // TODO: use of Entry might be more efficient
-        if let Some(existing) = env.map.get(&key) {
-            if !existing.is_same_type(&raw) {
-                return Err(ValueTypeError::new(any::type_name::<V>(), raw));
+        let entry = env.map.entry(key);
+        match entry {
+            std::collections::hash_map::Entry::Occupied(mut e) => {
+                let existing = e.get_mut();
+                if !existing.is_same_type(&raw) {
+                    return Err(ValueTypeError::new(any::type_name::<V>(), raw));
+                } else {
+                    *existing = raw;
+                }
```

Comment:
else is not required :)
```suggestion
                }
                *existing = raw;
```